akka-docs/src/main/paradox/stream/operators/Source-or-Flow/watchTermination.md

```diff
@@ -15,6 +15,16 @@ Materializes to a @scala[`Future`] @java[`CompletionStage`] that will be complet
 Materializes to a @scala[`Future`] @java[`CompletionStage`] that will be completed with Done or failed depending whether the upstream of the operators has been completed or failed.
 The operators otherwise passes through elements unchanged.
 
+## Examples
+
+Scala
+:   @@snip [WatchTermination.scala](/akka-docs/src/test/scala/docs/stream/operators/sourceorflow/WatchTermination.scala) { #watchTermination }
+
+Java
+:   @@snip [WatchTermination.java](/akka-docs/src/test/java/jdocs/stream/operators/SourceOrFlow.java) { #watchTermination } 
+
+The second parameter of this operator is a function that can be used to map the materialized value of the stream. Additionally, you could use it to clean up any resources used by the stream itself. 
```

Comment:
It is only the second parameter in the Scala API so maybe skip that. 

Additionally I think this can be misinterpreted to say that the cleanup can be put in the lambda itself, but that is run immediately on materialization, the completion of the future/completion stage can be used for cleanup, so please clarify that.

You can define different text for Scala and Java shown depending on which language the docs are browsed in like this: `@scala[Future]@java[CompletionStage]`

Reply:
Thanks for the review! I changed the description based on your suggestions.